test/jdk/java/lang/System/i18nEnvArg.java

```diff
@@ -0,0 +1,147 @@
+        /*
+         * Checks OS is Linux and OS has ja_JP.eucjp locale or not.
+         * Sets EUC_JP's environment variable and argunments against ProcessBuilder
+         */
+        public static void main(String[] args) throws Exception {
+            String nativeEncoding = System.getProperty("native.encoding");
+            Charset dcs = nativeEncoding == null ?
+                Charset.defaultCharset() :
+                Charset.forName(nativeEncoding);
+            Charset cs = Charset.forName("x-euc-jp-linux");
+            if (!dcs.equals(cs)) {
+                return;
+            }
+            String javeExe = System.getProperty("java.home") +
+                File.separator +
+                "bin" +
+                File.separator +
+                "java";
+            ProcessBuilder pb = new ProcessBuilder(javeExe,
+                "--add-opens=java.base/java.lang=ALL-UNNAMED",
+                "-classpath",
+                System.getProperty("java.class.path"),
+                "i18nEnvArg$Verify",
+                EUC_JP_TEXT);
+            pb.redirectErrorStream(true);
+            Map<String, String> environ = pb.environment();
+            environ.clear();
+            environ.put("LANG", "ja_JP.eucjp");
+            environ.put(EUC_JP_TEXT, EUC_JP_TEXT);
+            Process p = pb.start();
+            InputStream is = p.getInputStream();
+            byte[] ba = is.readAllBytes();
+            int rc = p.waitFor();
+            if (ba.length > 0)
+                throw new Exception(new String(ba));
+        }
+    }
+
+    public static class Verify {
+        private final static int maxSize = 4096;
+        /*
+         * Verify environment variable and argument are encoded by Linux's eucjp or not
+         */
+        public static void main(String[] args) throws Exception {
+            Charset cs = Charset.forName("x-euc-jp-linux");
+            byte[] euc = (EUC_JP_TEXT + "=" + EUC_JP_TEXT).getBytes(cs);
+            byte[] eucjp = "LANG=ja_JP.eucjp".getBytes(cs);
+            String s = System.getenv(EUC_JP_TEXT);
+            if (!EUC_JP_TEXT.equals(s)) {
+                System.out.println("getenv() returns unexpected data");
+            }
+            if (!EUC_JP_TEXT.equals(args[0])) {
+                System.out.print("Unexpected argument was received: ");
+                for(char ch : EUC_JP_TEXT.toCharArray()) {
+                   System.out.printf("\\u%04X", (int)ch);
+                }
+                System.out.print("<->");
+                for(char ch : args[0].toCharArray()) {
+                   System.out.printf("\\u%04X", (int)ch);
+                }
+                System.out.println();
+            }
+            Class<?> cls = Class.forName("java.lang.ProcessEnvironment");
+            Method environ_mid = cls.getDeclaredMethod("environ");
+            environ_mid.setAccessible(true);
+            byte[][] environ = (byte[][]) environ_mid.invoke(null,
+                (Object[])null);
```

Comment:
I am not sure I like this piece, invoking a package private method of `ProcessEnvironment` class. Can we simply verify the result of `System.getenv(EUC_JP_TEXT)`?

Reply:
I think 3 processes required for this testing.
1. Start test process on ja_JP.eucjp locale
2. Set the test data by encoder
3. Verify the encoded data by decoder

To verify the encoded data, I think I need to check the byte data.

Reply:
Do we need to verify the intermediate byte encoding? Could we simply compare the text given to environ.put() in Start process and System.getenv() in Verify process match?

Reply:
Hello @naotoj .
I think if 2nd process' encoder (like UTF-8) and 3rd process' decoder are same encoding, System.getenv() returns expected data.
Actually the testcase checks 3 parts on Verify process:
1. Expected environment variable is defined or not (it uses System.getenv())
2. Expected argument is received or not
3. Expected environment variable is encoded by proper encoding

When I ran this testcase with jdk18.0.1, I got following result:
```
Unexpected argument was received: \u6F22\u5B57<->\u7FB2\u221A\uFFFD
Unexpected environment variables: \xE6\xBC\xA2\xE5\xAD\x97\x3D\xE6\xBC\xA2\xE5\xAD\x97
```
It means 1st test was passed, 2nd and 3rd test were failed.
I don't think environment variable issue can be seen without 3rd test.
Please let me know if you find out another way.

Reply:
This part of the test is very brittle; I'm pretty sure it will fail on AIX that adds its own environment variables.
It should not fail if it finds the two entries it expects. It should ignore other entries.

I don't see what value it has over checking the entries from System.getEnv(), please elaborate.